__tests__/pet.test.js

```diff
@@ -0,0 +1,281 @@
+    });
+
+    it('if both of the above are true, return "I feel great!"', () => {
+        const pet = new Pet('Fido');
+        pet.fitness = 5;
+        pet.hunger = 3;
+        pet.checkUp();
+
+        expect(pet.checkUp()).toBe('I feel great!');
+    });
+
+    it('throws an error if the pet is not alive', () => {
+        const pet = new Pet('Fido');
+        pet.age = 30;
+        expect(pet.checkUp()).toBe('Your pet is no longer alive :(');
+    });
+    
+});
+
+
+describe('isAlive', () => {
+
+    const pet = new Pet('Fido');
+
+    it('if fitness is 0 or less, return false', () => {
+        pet.fitness = -3;
+        pet.hunger = 5;
+        pet.age = 10;
+
+        expect(pet.isAlive).toBeFalsy();
+    });
+
+    it('if hunger is 10 or more, return false', () => {
+        pet.fitness = 3;
+        pet.hunger = 12;
+        pet.age = 10;
+
+        expect(pet.isAlive).toBeFalsy();
+    });
+
+    it('if age is 30 or more, return false', () => {
+        pet.fitness = 3;
+        pet.hunger = 5;
+        pet.age = 32;
+
+        expect(pet.isAlive).toBeFalsy();
+    });
+
+    it('if none of above, return true', () => {
+        pet.fitness = 3;
+        pet.hunger = 5;
+        pet.age = 10;
+
+        expect(pet.isAlive).toBeTruthy();
+    });
+})
+
+describe('adoptChild', () => {
+
+    const parent = new Pet('Yeppi');
+    const child1 = new Pet('Happy');
+
+    it('adopt a child with children property of array \
+    where the first element is the child instance you passed', () => {
+        parent.adoptChild(child1);
+
+        expect(parent.children).toEqual([ { name: 'Happy', age: 0, hunger: 0, fitness: 10, children: [] } ]);
+    });
+
+    it('adopt two more children, resulting in three children', () => {
+        const child2 = new Pet('Doory');
+        const child3 = new Pet('Nyangi');
+        parent.adoptChild(child2);
+        parent.adoptChild(child3);
+
+        expect(parent.children).toEqual(
+            [ 
+                { name: 'Happy', age: 0, hunger: 0, fitness: 10, children: [] },
+                { name: 'Doory', age: 0, hunger: 0, fitness: 10, children: [] },
+                { name: 'Nyangi', age: 0, hunger: 0, fitness: 10, children: [] } 
+            ]);
+    });
+
+    it('increase first child Happy\'s age by 2', () => {
+        parent.children[0].growUp();
+        parent.children[0].growUp();
+
+        expect(parent.children[0].age).toEqual(2);
+    });
+
+    it('increase third child Nyangi\'s age by 1, take her to walk and feed her', () => {
+        parent.children[2].growUp();
+        parent.children[2].walk();
+        parent.children[2].feed();
+
+        expect(parent.children[2]).toEqual(
+            { name: 'Nyangi', age: 1, hunger: 2, fitness: 10, children: [] } 
+        );
+    });
+
+    it('when a dead pet adopts a child, throw an error', () => {
+        parent.age = 30;
+        const child4 = new Pet('Congi');
```

Comment:
As mentioned earlier you need not create a variable to store this instance of `Pet` in, since it is only used on line 230 you could call it directly in the `adoptChild` function